task1/views.py

```diff
@@ -0,0 +1,141 @@
+from django.contrib import messages
+from django.contrib.auth import authenticate, login, logout
+from django.contrib.auth.models import User
+from django.core.exceptions import PermissionDenied
+from django.http import HttpResponseNotFound, HttpResponseServerError
+from django.shortcuts import redirect, render
+from django.views import View
+
+from .forms import EditForm, LoginForm, SignupForm
+
+
+class HomeView(View):
+    template_name = "task1/home.html"
+
+    def get(self, request, *args, **kwargs):
+        if not request.user.is_authenticated and request.method == "GET":
+            print("not authenticated --->>> returned to login")
+            return redirect("login")
+        return render(request, self.template_name)
+
+
+class SignupView(View):
+    template_name = "task1/signup.html"
+    form_class = SignupForm
+
+    def get(self, request, *args, **kwargs):
+        form = self.form_class()
+        return render(request, self.template_name, {"form": form})
+
+    def post(self, request, *args, **kwargs):
+        if request.method == "POST":
+            form = self.form_class(request.POST)
+            try:
+                if form.is_valid():
+                    first_name = form.cleaned_data.get("first_name")
+                    last_name = form.cleaned_data.get("last_name")
+                    username = form.cleaned_data.get("username")
+                    password = form.cleaned_data.get("password")
+                    confirm_password = form.cleaned_data.get(
+                        "confirm_password",
+                    )
+
+                    if password != confirm_password:
+                        messages.error(request, "Password Does not match")
+                        return redirect("signup")
+                    user = User.objects.create_user(
+                        first_name=first_name,
+                        last_name=last_name,
+                        username=username,
+                        password=password,
+                    )
+                    user.save()
+                    return redirect("login")
+                else:
+                    messages.error(request, "Invalid Entry")
+                    return redirect("signup")
+            except PermissionDenied as e:
+                return HttpResponseNotFound(f"Not found {e}")
+            except Exception as e:
```

Comment:
can you explain why are you using `try except` here when you've handled logic in `if else`

Reply:
I have read that it's a good practice to include error handling to handle unexpected exceptions that might occur during execution.